tests/shardddl3/run.sh

```diff
@@ -679,6 +679,197 @@ function DM_103() {
     "clean_table" "optimistic"
 }
 
+function DM_104_CASE() {
+    run_sql_source1 "alter table ${shardddl1}.${tb1} add new_col1 int not null default 10;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} (id) values(1);"
+    run_sql_source1 "alter table ${shardddl1}.${tb1} add new_col2 int not null default 20;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} (id) values(2);"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} values(3);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} values(4);"
+    run_sql_source2 "alter table ${shardddl1}.${tb1} add new_col2 int not null default 20;"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} (id) values(5);"
+    run_sql_source2 "alter table ${shardddl1}.${tb1} add new_col1 int not null default 10;"
+    run_sql_source2 "insert into ${shardddl1}.${tb1} (id) values(6);"
+    run_sql_source2 "alter table ${shardddl1}.${tb2} add new_col1 int not null default 10;"
+    run_sql_source2 "alter table ${shardddl1}.${tb2} add new_col2 int not null default 20;"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} (id) values(7);"
+    run_sql_source1 "insert into ${shardddl1}.${tb1} (id) values(8);"
+    run_sql_source2 "insert into ${shardddl1}.${tb2} (id) values(9);"
+    run_sql_tidb_with_retry "select count(1) from ${shardddl}.${tb};" "count(1): 9"
+}
+
+function DM_104() {
+    # currently not support pessimistic
+    # run_case 104 "double-source-pessimistic" "init_table 111 211 212" "clean_table" "pessimistic"
+    run_case 104 "double-source-optimistic" "init_table 111 211 212" "clean_table" "optimistic"
+}
+
+function ADD_DROP_INDEX_CASE() {
```

Comment:
hope the helper function has a lower case name or something, to distinguish it from DM cases